starknet_py/abi/v1/parser_transformer.py

```diff
@@ -0,0 +1,120 @@
+import os
+from typing import Any, List, Optional
+
+import lark
+from lark import Token, Transformer, v_args
+
+from starknet_py.cairo.data_types import (
+    ArrayType,
+    CairoType,
+    FeltType,
+    OptionType,
+    TupleType,
+    TypeIdentifier,
+    UintType,
+)
+
+
+class ParserTransformer(Transformer):
+    """
+    Transforms the lark tree into CairoTypes.
+    """
+
+    # pylint: disable=no-self-use
+
+    def __default__(self, data: str, children, meta):
+        raise TypeError(f"Unable to parse tree node of type {data}")
+
+    @v_args(inline=True)
+    def start(self, value) -> Optional[CairoType]:
+        """
+        Method all the entries starts from.
+        """
+        return value
+
+    def type(self, value: List[Optional[CairoType]]) -> Optional[CairoType]:
+        """
+        Types starting with "core" or unit type "()".
+
+        Tokens are read bottom-up, so here all of them are parsed and should be just returned.
+        `Optional` is added in case of the unit type.
```

Comment:
Could you clarify in the docstring this sentence? What's the connection here between Optional and unit type?

Reply:
I think I understood after reading the rest of docstrings, though not so obvious at first

Reply:
I meant `Optional` in typing. `unit` type is returned as `None`